### PR TITLE
feat: drag n drop to reorder tasks

### PR DESCRIPTION
**Pull request by Chicarelli**
Implements #25 

Drag and drop feature to reorder tasks using react-beautiful-dnd

I put the icon before the task name because when I was testing, the layout shift seems better to me. But i'm not designer and I have a poor knowledge on tailwinds classes...

### CODE REVIEW

**Comment by YuriSamp** [comment]
do you have any idea to avoid this?

PhivedDragAndDrop

File: src/main.tsx
[the remark this answers is not in the document]
Comment (Chicarelli) [reply src/main.tsx:1]: react-beautiful-dnd doesn't work on dev environment with react 18+.  [Issue on react-bautiful-dnd](https://github.com/atlassian/react-beautiful-dnd/issues/2407)

Options are either remove strict mode or comment everytime we run locally.

**Comment by Chicarelli** [comment]
> do you have any idea to avoid this?
> 
> PhivedDragAndDrop

Can you tell how you got this? I'm draggind the tasks randomly and didn't get to replicate this

**Comment by LukeberryPi** [comment]
> do you have any idea to avoid this?
> 
> PhivedDragAndDrop

to my understanding, even though the user can drag a task outside of the Tasks area, it can't be dropped. as in it just snaps back to the expected area, which is expected behaviour. if the user could drop it outside, then we would have a problem.

**Comment by YuriSamp** [comment]
is just a aesthetic thing., if you drag a task outside the container, when you drop, it comeback to original place

**Comment by LukeberryPi** [comment]
> is just a aesthetic thing., if you drag a task outside the container, when you drop, it comeback to original place

we can keep it as is.

**Comment by Chicarelli** [comment]
I though the drag button was to appear in all cases, then the layout shift being beside the done button and appearing under the same conditions, I think it will be good.  I'll work on the adjustments and update de pr soon.

**Comment by LukeberryPi** [comment]
great stuff, thanks for addressing all that

we have only one issue before merging this, the rest works fine:

from 320px..767px there are rough corners on top and bottom right. i think it is derived from one of your added divs. for simplicity, you can just slap "rounded-2xl", which is the default border radius for the whole project.
image

**Review by LukeberryPi — APPROVED** [review]

**Comment by Chicarelli** [comment]
@LukeberryPi  I notice a border effect that have in production and also on my branch, that is this border: 
image

This algo happens on lastTask. I added classes to avoid this when task is not empty. Hope its ok

**Comment by LukeberryPi** [comment]
> @LukeberryPi I notice a border effect that have in production and also on my branch: image
> 
> This algo happens on lastTask. I added classes to avoid this when task is not empty. Hope its ok
> 
> **OBS:** In one of my monitors I can see cleary a border, in the other one, i can't. So if you can't see, believe me, there is a border lol

great catch! my eyesight isn't the best, so i definetely can't see that. thanks for the change

**Pull request merged**